src/common/libutil/intree.c

```diff
@@ -0,0 +1,106 @@
+/************************************************************\
+ * Copyright 2019 Lawrence Livermore National Security, LLC
+ * (c.f. AUTHORS, NOTICE.LLNS, COPYING)
+ *
+ * This file is part of the Flux resource manager framework.
+ * For details, see https://github.com/flux-framework.
+ *
+ * SPDX-License-Identifier: LGPL-3.0
+\************************************************************/
+
+#if HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include <stdlib.h>
+#include <unistd.h>
+#include <stdbool.h>
+#include <string.h>
+#include <errno.h>
+#include <sys/param.h> /* MAXPATHLEN */
+#include <libgen.h>    /* dirname(3) */
+#include <pthread.h>
+
+#include <flux/core.h>
+
+static pthread_mutex_t intree_lock = PTHREAD_MUTEX_INITIALIZER;
```

Comment:
if these variables don't need to be shared between the functions, consider moving them to function scope.

Reply:
Good point. In another iteration I was using TLS and those variables were shared. I'll make the change.